app/controllers/concerns/vaccination_mailer_concern.rb

```diff
@@ -59,8 +59,10 @@ def parents_for_vaccination_mailer(vaccination_record)
       ConsentGrouper.call(patient.consents, programme_id:, academic_year:)
 
     parents =
-      if consents.any?(&:via_self_consent?)
-        consents.any?(&:notify_parents_on_vaccination) ? patient.parents : []
+      if !vaccination_record.notify_parents
+        []
```

Comment:
I was thinking even above `ConsentGrouper.call`, since we're not using `consents` in this case. Perhaps something like:

```rb
return [] unless patient.send_notifications? && vaccination_record.notify_parents?
```